src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -116,79 +118,92 @@ func logHostsStatus(log logrus.FieldLogger, hosts map[string]inventory_client.Ho
 	log.Infof("Hosts status: %v", hostsStatus)
 }
 
-func (c *controller) WaitAndUpdateNodesStatus(status *ControllerStatus) {
+func (c *controller) WaitAndUpdateNodesStatus(ctx context.Context, wg *sync.WaitGroup) {
+	approveCtx, approveCancel := context.WithCancel(ctx)
+	defer func() {
+		approveCancel()
```

Comment:
This is a change from the previous logic where the CSR still lived 10 minutes after the node scan finished. And in OCP there is no timeout at all. Now it will be terminated exactly when the wait for node finishes. Is this done on purpose?

Reply:
This is what was expected. After all nodes are added there is no point to have approve csr up.